### PR TITLE
Convert directly from DOMString to Vec<u8>

### PR DESCRIPTION
**Pull request by saurvs**
This change is [Review on Reviewable](https://reviewable.io/reviews/servo/servo/9967)

### CODE REVIEW

**Comment by Ms2ger** [comment]
@bors-servo r+

Thank you!

File: components/script/dom/blob.rs [hunk 1]
```diff
@@ -124,8 +124,7 @@ impl Blob {
                         blobPropertyBag: &BlobBinding::BlobPropertyBag)
                         -> Fallible<Root<Blob>> {
         // TODO: accept other blobParts types - ArrayBuffer or ArrayBufferView or Blob
-        // FIXME(ajeffrey): convert directly from a DOMString to a Vec<u8>
-        let bytes: Vec<u8> = String::from(blobParts).into_bytes();
+        let bytes: Vec<u8> = blobParts.into();
```
Comment (KiChjang) [inline components/script/dom/blob.rs]: Do we need the type annotations at all for bytes?

Comment (saurvs) [inline components/script/dom/blob.rs]: I guess not. We don't need the variable either.

**Comment by jdm** [comment]
@bors-servo: retry
* #7846

**Pull request merged**